topostats/run_topostats.py

```diff
@@ -37,7 +48,47 @@
 # pylint: disable=too-many-nested-blocks
 
 
-def run_topostats(args: None = None) -> None:  # noqa: C901
+def handle_config_file(args: argparse.Namespace | None) -> dict:
```

Comment:
I'd say this function is more parsing arguments, one of which may be a configuration file. It reconciles these options against the defaults so perhaps...

```suggestion
def reconcile_config_args(args: argparse.Namespace | None) -> dict:
```

...might be more descriptive.

Reply:
I was very unhappy with the name I gave it so I'm happy that you've come up with a better one